Measure rwc per object

emiglietta opened the pull request.

emiglietta committed:
load measurementtemplate.py

emiglietta committed:
load all necessary parts

Now, I have to create the for loop function.
I deleted of commented all parts having to do with other measurements (including per image measurements) and also removed if statements, assuming here wants_obects is always True and perform_RWC is always True

emiglietta committed:
add debug configuration

emiglietta committed:
remove the remainder of the sample plugin

emiglietta committed:
renamed module

in order to escape import errors in CP (name of file must be all lower case) and debugger cache (Nodar had to change the actual name bc the previous capitalized one remained cached)

bethac07 commented on active_plugins/measureRWCperObject.py:
```diff
@@ -0,0 +1,524 @@
+        if len(self.objects_list.value) == 0:
+            raise ValidationError("No object sets selected", self.objects_list)
+
+    def upgrade_settings(self, setting_values, variable_revision_number, module_name):
```
You don't need this for this plugin (yet)

bethac07 commented on active_plugins/measureRWCperObject.py:
```diff
@@ -0,0 +1,524 @@
+M_PER_OBJECT = "Within each object individually"
+
+"""Feature name format for the RWC Coefficient measurement"""
+F_RWC_FORMAT = "Correlation_RWC_%s_%s"
```
This may cause a collision if the person has also measured RWC regularly, can we name it something different?

bethac07 commented on active_plugins/measureRWCperObject.py:
```diff
@@ -0,0 +1,524 @@
+        return True
+
+
+def get_scale(scale_1, scale_2):
```
Is this needed?

bethac07 commented on active_plugins/measureRWCperObject.py:
```diff
@@ -0,0 +1,524 @@
+#################################
+
+import centrosome.cpmorphology
+import centrosome.zernike
```
Make sure to remove the ones you don't. need

bethac07 commented on .vscode/launch.json:
```diff
@@ -0,0 +1,28 @@
+{
```
Can you remove this file from the commit, please?

bethac07 reviewed (changes requested):
Awesome job! Just some cleanup to do, plus add to the documentation

emiglietta committed:
Update measurerwcperobj.py

emiglietta committed:
looks on its way but i messed too much with the code before, might have

looks on its way but i messed too much with the code before, might have to restart fresh

emiglietta committed:
not fancy, but it works!!

emiglietta committed:
resolve conflict

emiglietta committed:
correct typo

emiglietta committed:
rename variables, working version!

still have to check that the values make sense

emiglietta committed:
cleanup and change F_RWC_FORMAT

emiglietta committed:
fix naming of results

emiglietta committed:
Update measurerwcperobj.py

add comments clarifying R_perObj and Di_perObj

emiglietta committed:
calculate threshold based on the max pixel intensity of each object individually

emiglietta committed:
correct thresholding per object using tff and tss.

Need to check why Correlation_RWCperObj_V5_ER gives inf values and values way above 1!

emiglietta committed:
fix thresholding per object

emiglietta committed:
implement threshold check

measure absolute and relative (vs the pixels of the object) above-threshold (both thresholds) pixels

emiglietta committed:
correct thresholding

I changed thresholding to be >= instead of >, bc otherwise, pixels of value 0 get automatically excluded no matter the threshold

emiglietta committed:
delete unused code, fix naming of RWC1/2 and go back to setting threshold as greater only

bethac07 commented:
@emiglietta Can you remove the extra file and do some cleaning of unneeded commented text, when you have a sec, so we can merge? Thanks!
